test/unit/alignment/configuration/align_config_gap_test.cpp

```diff
@@ -0,0 +1,195 @@
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \brief Provides tests for seqan3::align_config_gap.
+ * \author Jörg Winkler <j.winkler AT fu-berlin.de>
+ */
+
+#include <gtest/gtest.h>
+
+#include <functional>
+#include <type_traits>
+
+#include <seqan3/alignment/configuration/align_config_gap.hpp>
+#include <seqan3/core/detail/reflection.hpp>
+
+using namespace seqan3;
+
+struct bar
+{
+    int value;
+};
+
+TEST(align_config_gap, constructor)
+{
+    EXPECT_TRUE((std::is_default_constructible_v<detail::align_config_gap<int32_t>>));
+}
+
+TEST(align_config_gap, on_align_config)
+{
+    using gap_config_type = detail::align_config_gap<int32_t>;
+
+    EXPECT_TRUE((std::is_same_v<typename detail::on_align_config<align_cfg::id::gap>::invoke<gap_config_type>,
+                 std::true_type>));
+    EXPECT_TRUE((std::is_same_v<typename detail::on_align_config<align_cfg::id::gap>::invoke<bar>,
+                 std::false_type>));
+}
+
+TEST(align_config_gap, align_config_type_to_id)
+{
+    using gap_config_type = detail::align_config_gap<int32_t>;
+
+    EXPECT_EQ(detail::align_config_type_to_id<gap_config_type>::value, align_cfg::id::gap);
+    EXPECT_EQ(detail::align_config_type_to_id_v<gap_config_type>, align_cfg::id::gap);
+}
+
+TEST(align_config_gap, invoke_linear)
+{
+    {
+        auto cfg = std::invoke(align_cfg::gap(gap_score{-4}), detail::configuration<>{});
+
+        EXPECT_EQ(std::get<0>(cfg).value.get_gap_score(), -4);
+        EXPECT_TRUE((std::is_same_v<remove_cvref_t<decltype(cfg)>,
+                                    detail::configuration<detail::align_config_gap<gap_scheme<int32_t>>>>));
+    }
+
+    {
+        detail::configuration cfg = align_cfg::gap(gap_score{-4.1});
+
+        EXPECT_DOUBLE_EQ(std::get<0>(cfg).value.get_gap_score(), -4.1);
+        EXPECT_TRUE((std::is_same_v<remove_cvref_t<decltype(cfg)>,
+                                    detail::configuration<detail::align_config_gap<gap_scheme<double>>>>));
+    }
+}
+
+TEST(align_config_gap, invoke_affine)
+{
+    {
+        auto cfg = std::invoke(align_cfg::gap(gap_score{-2}, gap_open_score{-4}), detail::configuration<>{});
+
+        EXPECT_EQ(get<0>(cfg).value.get_gap_score(), -2);
+        EXPECT_EQ(get<0>(cfg).value.get_gap_open_score(), -4);
+        EXPECT_TRUE((std::is_same_v<remove_cvref_t<decltype(cfg)>,
+                                    detail::configuration<detail::align_config_gap<gap_scheme<int32_t>>>>));
+    }
+
+    {
+        detail::configuration cfg = align_cfg::gap(gap_score{-2.2f}, gap_open_score{-4.1f});
+
+        EXPECT_FLOAT_EQ(get<0>(cfg).value.get_gap_score(), -2.2f);
+        EXPECT_FLOAT_EQ(get<0>(cfg).value.get_gap_open_score(), -4.1f);
+        EXPECT_TRUE((std::is_same_v<remove_cvref_t<decltype(cfg)>,
+                                    detail::configuration<detail::align_config_gap<gap_scheme<float>>>>));
+    }
+}
```

Comment:
Can you also add a test for directly invoking with gap_scheme

Reply:
Sure, I added tests for gap_scheme and fixed the doc error.
I rebased on top of the master branch for the Jenkins performance test.